utam-compiler/src/main/java/utam/compiler/helpers/TypeUtilities.java

```diff
@@ -93,6 +109,52 @@ static boolean isParametersTypesMatch(List<TypeProvider> expectedTypes,
     return true;
   }
 
+  public static String[] processTypeNode(
```

Comment:
i'd put this into separate utilities, but it's more just personal preference

Reply:
I figured we're creating a type reference, so `TypeUtilities` is the best place. Suggestions where to relocate? `JsonDeserializer`?

Reply:
i would do it as separate file as TypeUtilities seem to be large, but then again it does seem appropriate place, so never mind, let's leave as is